### PR TITLE
feat: native token config

### PR DESCRIPTION
**Pull request by r4mmer**
### Acceptance Criteria
- Remove uses of HATHOR_TOKEN_CONFIG
- Remove all hardcoded mentions of HTR
- Use the native token configured by the fullnode (or use HTR as default)

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/components/token/TokenAutoCompleteField.js [hunk 1]
```diff
@@ -4,9 +4,14 @@ import Loading from '../Loading';
 import tokensApi from '../../api/tokensApi';
 import { debounce, get } from 'lodash';
 import { constants as hathorLibConstants } from '@hathor/wallet-lib';
+import { connect } from "react-redux";
 
 const DEBOUNCE_SEARCH_TIME = 200; // ms
 
+const mapStateToProps = (state) => {
+  return { serverInfo: state.serverInfo };
```
Comment (pedroferreira1) [inline src/components/token/TokenAutoCompleteField.js:12]: Why do you need this? Is it in use?

**Review by pedroferreira1 — APPROVED** [review]

File: src/components/tx/TxData.js [hunk 2]
```diff
@@ -119,6 +125,11 @@ class TxData extends React.Component {
     });
   }
 
+  getNativeToken = () => {
```
Comment (tuliomir) [inline src/components/tx/TxData.js:128]: question(non-blocking): Should we use `wallet.storage.getNativeTokenData()` instead? Or we don't have any wallets available in store?

**Review by tuliomir — APPROVED** [review]

File: src/components/token/TokenAutoCompleteField.js (hunk 1 above)
Comment (r4mmer) [reply src/components/token/TokenAutoCompleteField.js:12]: Nice catch, the placeholder in this file used "Hathor (HTR) ..." so I had to change it but I think I skipped this line.

File: src/components/token/TokenAutoCompleteField.js [hunk 3]
```diff
@@ -21,6 +26,14 @@ class TokenAutoCompleteField extends React.Component {
     this.handleClick = this._handleClick.bind(this);
   }
 
+  getNativeToken() {
+    return {
+      uid: hathorLibConstants.NATIVE_TOKEN_UID,
+      name: this.props.serverInfo?.native_token?.name ?? hathorLibConstants.DEFAULT_TOKEN_CONFIG.name,
+      symbol: this.props.serverInfo?.native_token?.symbol ?? hathorLibConstants.DEFAULT_TOKEN_CONFIG.symbol,
```
Comment (pedroferreira1) [inline src/components/token/TokenAutoCompleteField.js:33]:
```suggestion
      name: this.props.serverInfo?.native_token?.name ?? hathorLibConstants.DEFAULT_NATIVE_TOKEN_CONFIG.name,
      symbol: this.props.serverInfo?.native_token?.symbol ?? hathorLibConstants.DEFAULT_NATIVE_TOKEN_CONFIG.symbol,
```

**Review by pedroferreira1 — CHANGES_REQUESTED** [review]

**Review by tuliomir — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**